.circleci/config.yml

```diff
@@ -492,12 +493,13 @@ defaults:
 
   - base_osx: &base_osx
       macos:
-        xcode: "14.2.0"
-      resource_class: macos.x86.medium.gen2
+        xcode: "15.0.0"
+      resource_class: macos.m1.medium.gen1
```

Comment:
I'm a bit confused - we have a universal binary that can be run on both ARM and x86 (Intel) architectures - but we're running everything on an ARM machine in the CI - which means we're running the same binary twice (for generating the bytecode reports), but calling one x86 - even though they're both the same binary (arm)? Aside from the `binary_type` and `label` fields, everything else is the same in `b_bytecode_osx_x86_64` and `b_bytecode_osx` - and those fields are irrelevant anyway, since if we want to actually do the bytecode comparison tests between the two architectures, we'd have to have a separate `b_osx_x86` job that would use the the x86 resource class.

Either that, or I'm heavily misunderstanding how this works.

Reply:
We build universal binaries on that arm machine. Where we run all the tests for `arm` on `arm`. For the `x86_64` part we use the [`arch` command](https://www.unix.com/man-page/osx/1/arch/) `arch -64 <executable> <parameters>` to execute the `x86_64` portion of the universal binary to generate the bytecode reports.

Reply:
Oh! I just saw that I didn't passed the `intel` parameter :D

Reply:
Interesting, there seem to be two different intel architectures:
- x86_64   64-bit intel
- x86_64h  64-bit intel (haswell)

So I guess we need to compare the byte-code of both

Reply:
OK, so yeah, you can run x86 binaries 'natively' on ARM chips either via Rosetta or `arch` - of course, in the case of universal binaries, you have to explicitly state that you want to run the x86 binary.

Also, I'd ignore haswell and just use x86_64 intel (haswell is just a codename for a generation of intel's x86 chips).

Reply:
I added this, because the `arch` command seem to treat that slightly different as `x86_64` - they put haswell as a special architecture parameter `x86_64h`